gobblin-runtime/src/main/java/org/apache/gobblin/service/monitoring/FlowStatusGenerator.java

```diff
@@ -154,16 +154,23 @@ public List<FlowStatus> getFlowStatusesAcrossGroup(String flowGroup, int countPe
    * @return true, if any jobs of the flow are RUNNING.
    */
   public boolean isFlowRunning(String flowName, String flowGroup, long flowExecutionId) {
-    List<FlowStatus> flowStatusList = getLatestFlowStatus(flowName, flowGroup, 1, null);
+    List<FlowStatus> flowStatusList = jobStatusRetriever.getAllFlowStatusesForFlowExecutionsOrdered(flowName, flowGroup);
+
     if (flowStatusList == null || flowStatusList.isEmpty()) {
       return false;
-    } else {
-      FlowStatus flowStatus = flowStatusList.get(0);
+    }
+    // Iterating through all flow statuses to check the condition
+    for (FlowStatus flowStatus : flowStatusList) {
       ExecutionStatus flowExecutionStatus = flowStatus.getFlowExecutionStatus();
-      log.info("Comparing flow execution status with flowExecutionId: " + flowStatus.getFlowExecutionId() + " and flowStatus: " + flowExecutionStatus + " with incoming flowExecutionId: " + flowExecutionId);
-      // If the latest flow status is the current job about to get kicked off, we should ignore this check
-      return flowStatus.getFlowExecutionId() != flowExecutionId && !FINISHED_STATUSES.contains(flowExecutionStatus.name());
+      log.info("Comparing flow execution status with flowExecutionId: " + flowStatus.getFlowExecutionId()
```

Comment:
I believe this would bloat up the logs. Earlier since we were comparing against only latest, it was okay.

We can move the log below when we are returning true which would help us understand which flowExecution was running.

Reply:
Agree that this will definitely bloat the logs up, would prefer this either be a debug log or move it outside of the loop whether or not the job is running within the if statement

Reply:
addressed